tests/unit_tests/test_rating_service.py

```diff
@@ -343,3 +343,20 @@ async def test_game_rating_error_handled(rating_service, game_info, bad_game_inf
     service._logger.warning.assert_called()
     # second game: results have been saved.
     service._persist_rating_changes.assert_called_once()
+
+
+async def test_game_update_empty_resultset_raises_exception(semiinitialized_service):
+    service = semiinitialized_service
+    game_id = 1
+    player_id = 1
+    rating_type = RatingType.GLOBAL
+    rating_type_id = service._rating_type_ids[RatingType.GLOBAL]
+    old_ratings = {player_id: Rating(1000, 500)}
+    after_mean = 1234
+    new_ratings = {player_id: Rating(after_mean, 400)}
+    outcomes = {player_id: GameOutcome.VICTORY}
+
+    await service._persist_rating_changes(
+        game_id, rating_type, old_ratings, new_ratings, outcomes
+    )
+    service._logger.warning.assert_called()
```

Comment:
I don't think the logger is the right thing to be checking here. You want to check that no SQL exception is thrown by the previous function call, which means you don't need to assert anything. If an exception is thrown, the test will fail. You just need to make sure that this test is written such that it will fail without your changes.